tests/director/tests/bundle_instance_auth_per_scenario_test.go

```diff
@@ -0,0 +1,290 @@
+package tests
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"github.com/kyma-incubator/compass/tests/pkg/gql"
+	"github.com/kyma-incubator/compass/tests/pkg/ptr"
+	"github.com/kyma-incubator/compass/tests/pkg/tenant"
+	"github.com/kyma-incubator/compass/tests/pkg/testctx"
+	"github.com/kyma-incubator/compass/tests/pkg/token"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnassignResourceFromScenarioWhenThereIsExistingBundleInstanceAuth(t *testing.T) {
```

Comment:
A successful test where you remove the BundleInstanceAuth and then try to remove the scenario will be good as well. (Not needed for every way of scenario removing, just one possible option to verify that once credentials are removed you are no longer blocked).